### PR TITLE
Mbstyles, disables partials, enable multi-font search, fix labelling as a consequence

### PR DESCRIPTION
**Pull request by aaime**
The end result of the activity looks like this:

image

image

### CODE REVIEW

File: modules/library/render/src/main/java/org/geotools/renderer/style/FontCache.java [hunk 1]
```diff
@@ -252,4 +259,31 @@ public synchronized void resetCache() {
...
+                            .stream()
+                            .filter(f -> f.startsWith(name))
+                            .filter(
+                                    f -> { // leave out alterations, use base fonts
+                                        String lc = f.toLowerCase();
+                                        return !lc.contains(" bold") && !lc.contains(" italic");
+                                    })
+                            .sorted() // leave further altered fonts down the line, base ones first
+                            .collect(Collectors.toList());
+        }
```
Comment (tbarsballe) [inline modules/library/render/src/main/java/org/geotools/renderer/style/FontCache.java]: Should this put the result into alternatives before returning? As far as I can tell, nothing ever gets written to alternatives currently.

File: modules/library/render/src/test/java/org/geotools/renderer/lite/LetterConflictTest.java [hunk 2]
```diff
@@ -55,8 +55,8 @@
      *
      * <p>Build with mvn -P image.interactive
      */
-    static final boolean IMAGE_INTERACTIVE =
-            Boolean.getBoolean("org.geotools.image.test.interactive");
+    static final boolean IMAGE_INTERACTIVE = true;
+    // Boolean.getBoolean("org.geotools.image.test.interactive");
```
Comment (tbarsballe) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/LetterConflictTest.java]: This should be restored to its prior state before merging

**Review by tbarsballe — CHANGES_REQUESTED** [review]
A couple things need fixing - see comments inline.

I also think the Javadocs could maybe do with a slightly more detailed explanation of what "Font Alternatives" actually are - the tests give a good example but the javadocs themselves are kinda vague (and the FontAlternativesFunction doesn't explain it at all), but this is relatively minor.

Everything else looks good.

File: modules/library/render/src/main/java/org/geotools/renderer/style/FontCache.java (hunk 1 above)
Comment (aaime) [reply modules/library/render/src/main/java/org/geotools/renderer/style/FontCache.java]: Good catch, thanks!

File: modules/library/render/src/test/java/org/geotools/renderer/lite/LetterConflictTest.java (hunk 2 above)
Comment (aaime) [reply modules/library/render/src/test/java/org/geotools/renderer/lite/LetterConflictTest.java]: Yep

**Comment by aaime** [comment]
@tbarsballe review feedback (hopefully) addressed.

**Review by tbarsballe — APPROVED** [review]

**Comment by aaime** [comment]
Thanks! Merging.

**Pull request merged**